src/jit/codegeninterface.h

```diff
@@ -220,10 +220,21 @@ class CodeGenInterface
     {
         return m_cgFramePointerRequired;
     }
+
     void setFramePointerRequired(bool value)
     {
         m_cgFramePointerRequired = value;
     }
+
+    //------------------------------------------------------------------------
+    // resetWritePhaseForFramePointerRequired: Return m_cgFramePointerRequired into the write phase.
+    // It is used only before the first phase, that locks this value, currently it is LSRA.
+    // Use it if you want to skip checks that set this value to true if the value is already true.
+    void resetWritePhaseForFramePointerRequired()
```

Comment:
I worry about adding this "escape hatch". It seems like this is needed because the StackLevelSetter is reading the value so it can check previous work, but setting the value if it determines the FP is required, on its own (or maybe together with previous computation?) Does the phase need to do both things? Or should it just compute the value and write it? Or should it write some global that later contributes the the "real" setting of the value?

Reply:
> It seems like this is needed because the StackLevelSetter is reading the value so it can check previous work, 

Yes, but it is the first reason.
The second reason is to save us from wasting time in this phase if frame pointer is already known to be required.